File: Src/java/engine-fhir/src/main/java/org/opencds/cqf/cql/engine/fhir/retrieve/CodeFilter.java
package org.opencds.cqf.cql.engine.fhir.retrieve;

import org.opencds.cqf.cql.engine.runtime.Code;

public class CodeFilter {
    public CodeFilter(String codePath, Iterable<Code> codes, String valueSet) {
        this.codePath = codePath;
        this.codes = codes;
        this.valueSet = valueSet;
    }

    private String codePath;

    public String getCodePath() {
        return codePath;
    }

    private Iterable<Code> codes;

    public Iterable<Code> getCodes() {
        return codes;
    }

    private String valueSet;

    public String getValueSet() {
        return valueSet;
    }
}

File: Src/java/engine-fhir/src/main/kotlin/org/opencds/cqf/cql/engine/fhir/retrieve/CodeFilter.kt
package org.opencds.cqf.cql.engine.fhir.retrieve

import org.opencds.cqf.cql.engine.runtime.Code

class CodeFilter(
    @JvmField val codePath: String?,
    @JvmField val codes: Iterable<Code>?,
    @JvmField val valueSet: String?,
)
